x/auth/ante_test.go

```diff
@@ -273,6 +289,50 @@ func TestAnteHandlerFees(t *testing.T) {
 	assert.True(t, feeCollector.GetCollectedFees(ctx).IsEqual(sdk.Coins{sdk.NewCoin("atom", 150)}))
 }
 
+// Test logic around memo gas consumption.
+func TestAnteHandlerMemoGas(t *testing.T) {
+	// setup
+	ms, capKey, capKey2 := setupMultiStore()
+	cdc := wire.NewCodec()
+	RegisterBaseAccount(cdc)
+	mapper := NewAccountMapper(cdc, capKey, &BaseAccount{})
+	feeCollector := NewFeeCollectionKeeper(cdc, capKey2)
+	anteHandler := NewAnteHandler(mapper, feeCollector)
+	ctx := sdk.NewContext(ms, abci.Header{ChainID: "mychainid"}, false, nil, log.NewNopLogger())
+
+	// keys and addresses
+	priv1, addr1 := privAndAddr()
+
+	// set the accounts
+	acc1 := mapper.NewAccountWithAddress(ctx, addr1)
+	mapper.SetAccount(ctx, acc1)
+
+	// msg and signatures
+	var tx sdk.Tx
+	msg := newTestMsg(addr1)
+	privs, accnums, seqs := []crypto.PrivKey{priv1}, []int64{0}, []int64{0}
+	fee := NewStdFee(0, sdk.NewCoin("atom", 0))
+
+	// tx does not have enough gas
+	tx = newTestTx(ctx, msg, privs, accnums, seqs, fee)
+	checkInvalidTx(t, anteHandler, ctx, tx, sdk.CodeOutOfGas)
+
+	// tx with memo doesn't have enough gas
+	fee = NewStdFee(1001, sdk.NewCoin("atom", 0))
+	tx = newTestTxWithMemo(ctx, msg, privs, accnums, seqs, fee, "abcininasidniandsinasindiansdiansdinaisndiasndiadninsd")
+	checkInvalidTx(t, anteHandler, ctx, tx, sdk.CodeOutOfGas)
+
+	// memo too large
+	fee = NewStdFee(2001, sdk.NewCoin("atom", 0))
+	tx = newTestTxWithMemo(ctx, msg, privs, accnums, seqs, fee, "abcininasidniandsinasindiansdiansdinaisndiasndiadninsdabcininasidniandsinasindiansdiansdinaisndiasndiadninsdabcininasidniandsinasindiansdiansdinaisndiasndiadninsd")
+	checkInvalidTx(t, anteHandler, ctx, tx, sdk.CodeMemoTooLarge)
+
+	// tx has enough gas
+	fee = NewStdFee(1000, sdk.NewCoin("atom", 0))
+	tx = newTestTx(ctx, msg, privs, accnums, seqs, fee)
+	checkValidTx(t, anteHandler, ctx, tx)
```

Comment:
This doesn't test the inclusion of memo when it is of a correct size, and enough fees is provided. Such a test needs to show that the memo is included in the state.

Reply:
Added a test for inclusion of memo less than maximum size. State isn't really a property of the ante handler, that would make sense to test elsewhere.